Added docstring tag (#1686)
python-discord/bot: bot/resources/tags/docstring.md

> A [`docstring`](https://docs.python.org/3/glossary.html#term-docstring) is a string with triple quotes that often used in file, classes, functions, etc. Docstrings usually has clear explanation, parameter(s) and return type.

Reviewer: ```suggestion
A [`docstring`](https://docs.python.org/3/glossary.html#term-docstring) is a string with triple quotes that often used in file, classes, functions, etc. A docstring usually has clear explanation, parameter(s) and a return type.
```

> A [`docstring`](https://docs.python.org/3/glossary.html#term-docstring) is a string with triple quotes that often used in file, classes, functions, etc. Docstrings usually has clear explanation, parameter(s) and return type.

Reviewer: We should also comment on best practices in regards to what content should be in the explanation.

> Here's an example of usage of a docstring:

Reviewer: ```suggestion
Here's an example of a docstring:
```

```py
def greet(name, age) -> str:
  """
  :param name: The name to greet.
  :type name: str
  :param age: The age to display.
  :type age: int
  :return: String of the greeting.

>   """

Reviewer: You should include an explanation of this function in the example, to show best practice

  return_string = f"Hello, {name} you are {age} years old!"
  return return_string
```

> You can get the docstring by using `.__doc__` attribute. For the last example you can get it through: `print(greet.__doc__)`.

Reviewer: I'd suggest adding some further reading here at the bottom, with links to more detailed guides. Maybe this for example https://realpython.com/documenting-python-code/#docstrings-background

> A [`docstring`](https://docs.python.org/3/glossary.html#term-docstring) is a string with triple quotes that often used in file, classes, functions, etc. A docstring usually has clear explanation (such as what the function do, purposes of the function, and other details of the function), parameter(s) and a return type.

Reviewer: ```suggestion
A [`docstring`](https://docs.python.org/3/glossary.html#term-docstring) is a string with triple quotes that often used in file, classes, functions, etc. A docstring should have a clear explanation of exactly what the function does. You can also include descriptions of the function's parameter(s) and it's return type, as shown below.
```

Here's an example of a docstring:
```py
def greet(name, age) -> str:
  """

>   Greet someone with their name and age.

Reviewer: ```suggestion
  Return a string that greets the given person, including their name and age.
```
Actually document what the function does

  :param name: The name to greet.
  :type name: str
  :param age: The age to display.
  :type age: int
  :return: String of the greeting.
  """
  return_string = f"Hello, {name} you are {age} years old!"
  return return_string
```
You can get the docstring by using `.__doc__` attribute. For the last example you can get it through: `print(greet.__doc__)`.

> For more details about what docstring is and it's usage check out [https://realpython.com/documenting-python-code/#docstrings-background](realpython) or [https://www.python.org/dev/peps/pep-0257/#what-is-a-docstring](official PEP docs).

Reviewer: ```suggestion
For more details about what docstring is and it's usage check out this guide by [Real Python](https://realpython.com/documenting-python-code/#docstrings-background), or the [PEP-257 docs](https://www.python.org/dev/peps/pep-0257/#what-is-a-docstring).
```
This is how you do hyperlinks.
